modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/PodcastManagerImpl.kt

```diff
@@ -627,8 +628,31 @@ class PodcastManagerImpl @Inject constructor(
         return podcastDao.findAutoAddToUpNextPodcasts()
     }
 
-    override suspend fun refreshPodcastFeed(podcastUuid: String): Boolean {
-        return refreshServiceManager.refreshPodcastFeed(podcastUuid).isSuccessful
+    override suspend fun refreshPodcastFeed(podcast: Podcast): Boolean {
+        var response = refreshServiceManager.updatePodcast(
+            podcastUuid = podcast.uuid,
+            lastEpisodeUuid = podcast.latestEpisodeUuid,
+        )
+        Timber.i("Refresh podcast feed: %d", response.code())
+
+        while (response.code() == 202) {
+            val location = response.headers()["Location"]
+            val retryAfter = response.headers()["Retry-After"]?.toIntOrNull()
+            if (location != null && retryAfter != null) {
+                delay(retryAfter * 1000L)
+                response = refreshServiceManager.pollUpdatePodcast(location)
+                Timber.i("Refresh podcast feed poll: %d", response.code())
```

Comment:
This was my bad, can we change this to LogBuffer so we can debug users' issues. I wonder if we should add some new Tracks events in separate PRs.